sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/FileScan.scala

```diff
@@ -86,7 +86,7 @@ trait FileScan extends Scan
 
   override def equals(obj: Any): Boolean = obj match {
     case f: FileScan =>
-      fileIndex == f.fileIndex && readSchema == f.readSchema
+      fileIndex == f.fileIndex && readSchema == f.readSchema &&
```

Comment:
This change is not related to this PR, but it looks like a `&&` is missing from here.

Reply:
Nice catch. It seems that https://github.com/apache/spark/pull/27112 introduced this. 
 cc @dongjoon-hyun if we want to backport this to `3.1` and `3.0`.

Reply:
Oh... nice catch. cc: @gengliangwang , too.

Reply:
Credit goes to @bersprockets for catching this in SPARK-33482.

Reply:
wow ..